v7.9 Changelog (#6246)
mattermost/docs: source/install/self-managed-changelog.md

Latest Mattermost Releases:

- [Release v7.9 - Feature Release](#release-v7-9-feature-release)
- [Release v7.8 - Extended Support Release](#release-v7-8-extended-support-release)
- [Release v7.7 - Feature Release](#release-v7-7-feature-release)
- [Release v7.1 - Extended Support Release](#release-v7-1-extended-support-release)

## Release v7.9 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**Release day: March 16th, 2023**

### Important Upgrade Notes

 - Added a new index on ``Posts(OriginalId)``. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes 98.51s on MYSQL and 2.6s on PostgreSQL. To avoid any table locking, servers using PostgreSQL are recommended to create the index online before performing the upgrade using the ``CONCURRENTLY`` option, as in ``CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);``.

> 

Reviewer: - Disable global header. Admins can set the feature flag ``MM_FEATUREFLAGS_GLOBALDRAFTS`` to ``false`` to disable server-wide. It can't be disabled on a per-user basis. The code is located [here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L11).
- Disable insights. Admins can set the feature flag ``MM_FEATUREFLAGS_INSGIHTSENABLED`` to ``false`` to disable server-wide. It can't be disabled on a per-user basis. The code is located [here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L11).

Latest Mattermost Releases:

- [Release v7.9 - Feature Release](#release-v7-9-feature-release)
- [Release v7.8 - Extended Support Release](#release-v7-8-extended-support-release)
- [Release v7.7 - Feature Release](#release-v7-7-feature-release)
- [Release v7.1 - Extended Support Release](#release-v7-1-extended-support-release)

## Release v7.9 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**Release day: March 16th, 2023**

### Important Upgrade Notes

>  - Added a new API endpoint ``GET api/v4/posts/[POST_ID]/edit_history``. Users can run "CREATE INDEX CONCURRENTLY" before doing the upgrade to allow PostgreSQL to create the index without locking the table.

Reviewer 2: I just updated that PR description with times for the index creation. Can we use this?

>Added a new API endpoint `GET api/v4/posts/[POST_ID]/edit_history`.
>Added a new index on `Posts(OriginalId)`. For a database with 11.8 million posts, on a machine with a i7-11800H CPU (8 cores, 16 threads), 32GiB of RAM and SSD, the index creation takes:
>- MySQL: 98.51s
>- PostgreSQL: 2.6s
>
> To avoid any table locking, servers using Postgres are recommended to create the index online before performing the upgrade using the `CONCURRENTLY` option, as in 
>```CREATE INDEX CONCURRENTLY idx_posts_original_id ON Posts(OriginalId);```

 - Fixed an issue where the true-up notification in the invite modal did not render the call-to-action correctly.
 - Fixed new teams to use the updated translation for default channels after a configuration change.
 - Fixed a layout issue in the System Console for smaller-sized tablets.
 - Fixed an issue where a "plugin configured with a nil SecureConfig" warning was logged when starting each plugin.
 - Fixed an issue where portal availability was checked when not on Enterprise edition.
 - Fixed an issue where C# syntax highlighting was not working.
 - Fixed an issue where incoming webhooks changed the user's activity while the user was offline/away.
 - Fixed an issue where usernames were not clickable in the right-hand side.

### config.json
Multiple setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.

#### Changes to Team Edition and Enterprise Edition:
 - Added a new config setting section ``ProductSettings``.

### API Changes

>  - Added a new API endpoint ``GET api/v4/posts/[POST_ID]/edit_history``.
>  - Added an ``exclude_files_count`` parameter to exclude file counts from the channel stats API.
>  - Added a new API endpoint ``GET api/v4/posts/[POST_ID]/edit_history``.

Reviewer 2: This seems to be duplicated
```suggestion
 - Added a new API endpoint ``GET api/v4/posts/[POST_ID]/edit_history``.
 - Added an ``exclude_files_count`` parameter to exclude file counts from the channel stats API.
```

## Release v7.9 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**Release day: March 16th, 2023**

### Important Upgrade Notes

 - Added a new API endpoint ``GET api/v4/posts/[POST_ID]/edit_history``. Users can run "CREATE INDEX CONCURRENTLY" before doing the upgrade to allow PostgreSQL to create the index without locking the table.
 - Boards is served as an in-built product from within the Mattermost server instead of a plugin. If you want to disable this, please set the ``MM_FEATUREFLAGS_BoardsProduct`` environment variable to ``false``. While running in product mode, the boards plugin will remain disabled.

**IMPORTANT:** If you upgrade from a release earlier than v7.8, please read the other [Important Upgrade Notes](/upgrade/important-upgrade-notes.html).

### Compatibility
 - Updated Firefox minimum supported version to 102+.
 - Updated Safari minimum supported version to 16.2+.
 - Updated Windows minimum supported version to 10+.
 - Updated Chromium minimum supported version to 110+.
 - Updated Edge minimum supported version to 110+.

### Highlights

#### Calls
 - 

> #### Boards

Reviewer 3: ```suggestion
#### Boards
```- System and team admins are now able to join any board on the team as a board admin via the board URL.

 - 
 
#### Playbooks
 - 

### Improvements

#### User Interface (UI)
 - All post components were removed in favor of a unified approach.
 - App bindings are now refreshed when a App plugin enabled event gets triggered.
 - Improvements were added to the sidebar channel and category menus.
 - Removed right-click hijacking on code blocks in messages.
 - The order of the Leave Channel and Archive Channel settings were updated to match the mobile app.
 - Added the condition to remove unread styling for archived channels and to filter archived channels from local data.
 - Changed the collapsed post fade out effect to be less buggy.
 - Users now have the ability to see the history of edited messages and to restore an old message version with the current version.
 - Improved the user interface of the user profile popover.

#### Administration
 - Boards cards are no longer mentioned as being limited in the **System Console**, the limits usage modal, the downgrade modal, or the left-hand side menu.

>  - Removed an unused ``ProductLimits.Integrations``.

Reviewer 4: ```suggestion
 - Removed unused ``ProductLimits.Integrations``.
```

### Go Version
 - v7.9 is built with Go ``v1.19.0``.

### Known Issues
 - Your profile image moves up when changing your status manually [MM-49159](https://mattermost.atlassian.net/browse/MM-49159).
 - The new Insights feature has some performance costs that we are working to optimize. This feature can be disabled by setting the ``MM_FEATUREFLAGS_INSIGHTSENABLED`` environment variable to ``false``.
 - Adding an @mention at the start of a post draft and pressing the left or right arrow key can clear the post draft and the undo history [MM-33823](https://mattermost.atlassian.net/browse/MM-33823).
 - Google login fails on the Classic mobile apps.
 - Status may sometimes get stuck as **Away** or **Offline** in High Availability mode with IP Hash turned off.
 - Searching stop words in quotation marks with Elasticsearch enabled returns more than just the searched terms.
 - The team sidebar on the desktop app does not update when channels have been read on mobile.
 - Slack import through the CLI fails if email notifications are enabled.
 - Push notifications don't always clear on iOS when running Mattermost in High Availability mode.

>  - Boards aren't refreshing on creation. See the [GitHub discussion](https://github.com/mattermost/focalboard/discussions/1971) for more information.

Reviewer 3: ```suggestion
```

 - Boards export and reimport results in duplicates boards because all IDs are replaced by new ones on the server. See the [GitHub issue](https://github.com/mattermost/focalboard/issues/1924) for more information.

>  - Boards linked to a channel you're a member of don't automatically appear on your sidebar unless you're an explicit member of the board. As a workaround, you can access the board from the channel RHS, or by searching for the board via the board switcher (Ctrl/Cmd+K). Alternatively, you can ask the board admin to add you to the board as an explicit member. See the [issue-focalboard-4179](https://github.com/mattermost/focalboard/issues/4179) for more details.

Reviewer 3: ```suggestion
```

>  - The Playbooks left-hand sidebar does not update when a user is added to a run or playbook without a refresh.

Reviewer 4: ```suggestion
 - The Playbooks left-hand sidebar doesn't update when a user is added to a run or playbook without a refresh.
```

 - Boards is served as an in-built product from within the Mattermost server instead of a plugin. If you want to disable this, please set the ``MM_FEATUREFLAGS_BoardsProduct`` environment variable to ``false``. While running in product mode, the boards plugin will remain disabled.

**IMPORTANT:** If you upgrade from a release earlier than v7.8, please read the other [Important Upgrade Notes](/upgrade/important-upgrade-notes.html).

### Compatibility
 - Updated Firefox minimum supported version to 102+.
 - Updated Safari minimum supported version to 16.2+.
 - Updated Windows minimum supported version to 10+.
 - Updated Chromium minimum supported version to 110+.
 - Updated Edge minimum supported version to 110+.

### Highlights

#### Calls
 - 

#### Boards

>  - System and team admins are now able to join any board on the team as a board admin via the board URL.

Reviewer 3: ```suggestion
 - System and team admins are now able to join any board on the team as a board admin via the board URL.
 - Additional Compliance APIs to return the history of boards and blocks, including deleted items (available in Mattermost Enterprise Edition and above).
```

 - Changed the collapsed post fade out effect to be less buggy.
 - Users now have the ability to see the history of edited messages and to restore an old message version with the current version.
 - Improved the user interface of the user profile popover.

#### Administration
 - Boards cards are no longer mentioned as being limited in the **System Console**, the limits usage modal, the downgrade modal, or the left-hand side menu.
 - Removed unused ``ProductLimits.Integrations``.
 - Export files now contain the read and unread status for channels.
 - While upgrading a High Availability installation with rolling upgrades, after upgrading all nodes, the System Console may display a single disabled plugin. This is expected due to the way plugins are initialized. Prepackaged plugins are copied over to the plugins directory. Following another restart of the first node upgraded, the System Console should no longer show any plugins present or enabled.
 - Added an error message when running an LDAP sync with ``SyncEnabled`` set to ``false``.
 - Added Admin log table filtering and sorting.
 - GraphQL APIs are now correctly counted when measuring performance telemetry.

>  - Added a dynamic call-to-action under the **System Console > True Up Review System** subsection for air-gapped and non-air-gapped systems.

Reviewer 5: Actually, the true-up CTA appears on System Console > Site Statistics and > Team Statistics

The banner reminding them about true-up follows the schedule outlined here:
https://docs.mattermost.com/about/self-hosted-subscriptions.html#quarterly-true-up-report
